src/pybind/mgr/dashboard/plugins/feature_toggles.py

```diff
@@ -0,0 +1,143 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+
+from enum import Enum
+import cherrypy
+from mgr_module import CLICommand, Option
+
+from . import PLUGIN_MANAGER as PM
+from . import interfaces as I
+from .ttl_cache import ttl_cache
+
+from ..controllers.rbd import Rbd, RbdSnapshot, RbdTrash
+from ..controllers.rbd_mirroring import (
+    RbdMirroringSummary, RbdMirroringPoolMode, RbdMirroringPoolPeer)
+from ..controllers.iscsi import Iscsi, IscsiTarget
+from ..controllers.cephfs import CephFS
+from ..controllers.rgw import Rgw, RgwDaemon, RgwBucket, RgwUser
+
+
+class Features(Enum):
+    RBD = 'rbd'
+    MIRRORING = 'mirroring'
+    ISCSI = 'iscsi'
+    CEPHFS = 'cephfs'
+    RGW = 'rgw'
+
+
+PREDISABLED_FEATURES = set()
+
+
+Feature2Controller = {
+    Features.RBD: [Rbd, RbdSnapshot, RbdTrash],
+    Features.MIRRORING: [
+        RbdMirroringSummary, RbdMirroringPoolMode, RbdMirroringPoolPeer],
+    Features.ISCSI: [Iscsi, IscsiTarget],
+    Features.CEPHFS: [CephFS],
+    Features.RGW: [Rgw, RgwDaemon, RgwBucket, RgwUser],
+}
+
+
+class Actions(Enum):
+    ENABLE = 'enable'
+    DISABLE = 'disable'
+    STATUS = 'status'
+
+
+@PM.add_plugin
+class FeatureToggles(I.CanMgr, I.CanLog, I.Setupable, I.HasOptions,
+                     I.HasCommands, I.FilterRequest.BeforeHandler,
+                     I.HasControllers):
+    OPTION_FMT = 'FEATURE_TOGGLE_{}'
+    CACHE_MAX_SIZE = 128  # Optimum performance with 2^N sizes
+    CACHE_TTL = 10  # seconds
+
+    @PM.add_hook
+    def setup(self):
+        url_prefix = self.mgr.get_module_option('url_prefix')
```

Comment:
'url_prefix' is not used.

Reply:
Thanks for the catch! I changed the initial implementation, which relied on URL matching, with this one, which uses controllers instead.